spec/src/main/asciidoc/ch03-entity-operations.adoc

```diff
@@ -1480,24 +1479,48 @@ An entity or entity listener may have multiple lifecycle callback methods:
 
 ==== Semantics of the Life Cycle Callback Methods for Entities [[a2202]]
 
-The `@PrePersist` and `@PreRemove` callbacks occur before the
-respective persist and remove operations for an entity are executed.
-For entities to which the `merge()` operation has been applied and
-causes the creation of newly managed instances, the `@PrePersist`
-callback occurs for the managed instance after the entity state has
-been copied to it. The `@PrePersist` and `@PreRemove` callbacks also
-occur for any entity to which these operations cascade. The `@PrePersist`
-and `@PreRemove` methods are always invoked as part of the synchronous
-`persist()`, `merge()`, and `remove()` operations of `EntityManager`.
-
-The `@PostPersist` and `@PostRemove` callbacks occur after an entity
-has been made persistent, or after it has been removed, respectively.
-These callbacks also occur for any entity to which these operations
-cascade. The `@PostPersist` and `@PostRemove` methods occur after the
-database insert and delete operations, respectively. The database
-operations might occur directly after invocation of `persist()`,
-`merge()`, or `remove()`, or they might occur during a `flush()`
-operation--which may not happen until the end of the transaction.
+A `@PrePersist`, `@PreRemove`, or `@PreMerge` callback occurs before
+the respective persist, remove, or merge operation for an entity is
+executed.
+
+- The `@PrePersist` callback for an entity being made persistent
+  occurs before the entity is marked as managed and associated with
+  the persistence context.
+- The `@PreRemove` callback occurs before the entity being removed
+  is marked as removed in the persistence context.
+- The `@PreMerge` callback for an entity being merged occurs before
+  state is copied from the entity being merged to the corresponding
+  managed entity, when copying is necessary.
+- When the `merge()` operation results in the creation of a new
+  managed instance, the `@PrePersist` callback for the managed
+  instance occurs after the merged entity state has been copied to
+  it.
+
+A `@PrePersist`, `@PreRemove`, or `@PreMerge` methods is always
+invoked synchronously during execution of the `persist()`, `merge()`,
+or `remove()` operation of the `EntityManager`.
```

Comment:
Easier to read if the same ordering is used in both parts of the sentence,
```suggestion
A `@PrePersist`, `@PreRemove`, or `@PreMerge` method is always
invoked synchronously during execution of the `persist()`, `remove()`,
or `merge()` operation of the `EntityManager`.
```

Reply:
This comment still applies - fixes a typo and suggests making the ordering of both parts of the sentence match